### DISCUSSION TITLE
RedisRequestLogger in RequestLogsFeature does not support NamespacePrefix

### DISCUSSION THREAD

**Discussion by LEIRONGHUA** [ServiceStack/Discuss#60] [Q&A] [answered]
According to the online documentation [https://docs.servicestack.net/request-logger#redis-request-logger](https://docs.servicestack.net/request-logger#redis-request-logger) settings, but Logging No Results

**Environment：**

- Redis Version：7.0.9
- ServiceStack.Server Version：6.8.0 ~ 6.8.1
- ServiceStack.Redis Version：6.8.0 ~ 6.8.1
- AspNetCore  Version：7.0

**Example code/Steps to reproduce:**

```
//Register your ServiceStack AppHost as a .NET Core module
app.UseServiceStack(new AppHost
{
    AppSettings = new NetCoreAppSettings(Configuration) // Use **appsettings.json** and config sources
});

// ConfigureRedis.cs
public void Configure(IWebHostBuilder builder)
{
    builder.ConfigureServices((context, services) =>
    {
        const string key = "ServiceStackRedisConnection";
        var redisHost = context.Configuration.GetConnectionString(key) ?? throw new ArgumentNullException(key);
        services.AddSingleton<IRedisClientsManager>(new BasicRedisClientManager(redisHost));
    });
}

// ConfigureProfiling.cs
public void Configure(IWebHostBuilder builder)
{
    builder.ConfigureAppHost(host =>
    {
        host.Plugins.Add(new RequestLogsFeature
        {
            EnableResponseTracking = true, // 启用响应跟踪
            EnableRequestBodyTracking = true, // 启用请求体跟踪
            RequestLogger = new RedisRequestLogger(host.Container.Resolve<IRedisClientsManager>(), capacity: 1000)
        });

        host.Plugins.Add(new ProfilingFeature
        {
            IncludeStackTrace = true,
        });
    });
}

// appsettings.json
{
    "ConnectionStrings": {
        "ServiceStackRedisConnection": "redis://127.0.0.1:6379?db=12&NamespacePrefix=ServiceStack%3A"
    }
}
```

**Output:**

```
Command:
MGET urn:requestlogentry:3 urn:requestlogentry:2 urn:requestlogentry:1

Logging:
No Results
```
![image](https://user-images.githubusercontent.com/4002277/233825281-fe58be5c-d8c5-412a-a62e-f0178d6c0826.png)
![image](https://user-images.githubusercontent.com/4002277/233825308-5979241e-bd10-4fee-8c77-6dea2b8097d0.png)
![image](https://user-images.githubusercontent.com/4002277/233825342-01e3ae2e-15ae-43d2-b80f-e1599c060237.png)

**Bug Analysis:**

The ServiceStack.Redis connection string sets the NamespacePrefix to "ServiceStack:", but the Log method of the ServiceStack.Host.RedisRequestLogger.cs class does not process the NamespacePrefix. So the cache key saved in Redis's Zset is wrong, for example: "urn:requestlogentry:1" is wrong, the correct one should be: "ServiceStack:urn:requestlogentry:1"
Please fix this bug, thanks!

**Comment by mythz** [comment, marked as answer]
I've changed it to use `redis.UrnKey<T>` in [this commit](https://github.com/ServiceStack/ServiceStack/commit/6fc31ad34062c7b409cf6a3e43d582bc8b333624) to be more consistent, but `NamespacePrefix` does not provide "true" keyspace partitioning and any change to existing behavior can result in breaking changes, so we'll be reluctant in making changes where there's a potential for breaking changes. You can submit PRs to propose changes which we can evaluate on a per change basis.

Anyway this change is available in the [latest v6.8.1 on MyGet](https://docs.servicestack.net/myget#redownloading-myget-packages).
